source/Calamari.Tests/KubernetesFixtures/TestableKubernetesDeploymentCommand.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using Calamari.Common.Commands;
+using Calamari.Common.Features.Packages;
+using Calamari.Common.Features.StructuredVariables;
+using Calamari.Common.Features.Substitutions;
+using Calamari.Common.Plumbing.Deployment.Journal;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment.Conventions;
+using Calamari.Kubernetes.Commands;
+using Calamari.Kubernetes.Integration;
+
+namespace Calamari.Tests.KubernetesFixtures
+{
+    [Command(Name, Description = "used for tests")]
+    public class TestableKubernetesDeploymentCommand : KubernetesDeploymentCommandBase
```

Comment:
I don't love that we have a different command for testing, even if the bulk of the logic in contained in the base class.
Is this because GatherAndApplyRawYamlConvention and ResourceStatusReportConvention can't be used in a test?

Reply:
This is only used to test the Auth components of the command base. There are still tests which do full deployments as well using the real `KubernetesApplyRawYamlCommand`. This is the equivalent of running the kubernetes script but not using the real deployment script, instead only using a dummy script. You can see that in tests like `AuthorisingWithAmazonAccount`, `AuthorisingWithPodServiceAccountToken`, `AuthorisingWithAzureServicePrincipal` etc. In those cases when using the script it will use a dummy script of `kubectl cluster-info` instead of the long complicated KubernetesDeployment.sh/ps1 scripts.

Reply:
Cool.  As long as we have test coverage of the real command, I'm good 👍

Reply:
Yep definitely, I just didn't want to do full deployments for every auth type as it's not really required so I tried to make an equivalent to a dummy script.